wled00/FX_fcn.cpp

```diff
@@ -223,12 +223,7 @@ CRGBPalette16 IRAM_ATTR &Segment::loadPalette(CRGBPalette16 &targetPalette, uint
     case 1: {//periodically replace palette with a random one
       unsigned long timeSinceLastChange = millis() - _lastPaletteChange;
       if (timeSinceLastChange > randomPaletteChangeTime * 1000U) {
-        _randomPalette = _newRandomPalette;
-        _newRandomPalette = CRGBPalette16(
-                        CHSV(random8(), random8(160, 255), random8(128, 255)),
-                        CHSV(random8(), random8(160, 255), random8(128, 255)),
-                        CHSV(random8(), random8(160, 255), random8(128, 255)),
-                        CHSV(random8(), random8(160, 255), random8(128, 255)));
+        _newRandomPalette = generateRandomPalette(_randomPalette);
```

Comment:
if not passing the base palette, the code for getting one random color needs to be in a separate function. did this for ease of use.

Reply:
could do that, but it is less accurate, as there are only 4 positions passed, so it will take an in-between color and shift that. but maybe good enough? not a bad Idea and much simpler. I will try it.